source/server/connection_handler_impl.cc

```diff
@@ -434,6 +448,11 @@ ConnectionHandlerImpl::ActiveTcpConnection::ActiveTcpConnection(
 }
 
 ConnectionHandlerImpl::ActiveTcpConnection::~ActiveTcpConnection() {
+  stream_info_->onRequestComplete();
+  for (const auto& access_log : config_.accessLogs()) {
+    access_log->log(nullptr, nullptr, nullptr, *stream_info_);
+  }
```

Comment:
Can this happen in deferred deletion context? Is this still safe?

Reply:
I think it should be fine. Stream info is a unique pointer owned by this object and access logs are  shared pointers.